gnd/src/main/java/com/google/android/gnd/persistence/remote/FirestoreStorageManager.java

```diff
@@ -61,34 +55,27 @@ private StorageReference getRootMediaDir() {
    * @param fileName Name of the uploaded media
    */
   private StorageReference createReference(String fileName) {
-    return getRootMediaDir().child(fileName + '-' + getFilenameSuffix());
+    return getRootMediaDir().child(fileName);
   }
 
-  /** Converts current timestamp to a string to be used a suffix for uploading media. */
-  private String getFilenameSuffix() {
-    return dateFormat.format(new Date());
+  public Maybe<Uri> getDownloadUrl(String path) {
+    return Maybe.create(
+        emitter ->
+            getBaseReference()
+                .child(path)
+                .getDownloadUrl()
+                .addOnSuccessListener(emitter::onSuccess)
+                .addOnFailureListener(emitter::onError));
   }
 
   /** Upload file to Firebase Storage. */
-  public void uploadMediaFromFile(File file, String fileName) {
+  public String uploadMediaFromFile(File file, String fileName) {
```

Comment:
What's the difference between `file` and `fileName`? Please rename to something self explanatory and/or explain in JavaDoc.